local/blockchain.go

```diff
@@ -440,13 +440,15 @@ func (ln *localNetwork) restartNodesWithWhitelistedSubnets(
 	zap.L().Info("restarting all nodes to whitelist subnet",
 		zap.Strings("whitelisted-subnets", whitelistedSubnetIDs),
 	)
+
+	// change default setting
+	ln.flags[config.WhitelistedSubnetsKey] = whitelistedSubnets
+
 	for nodeName, node := range ln.nodes {
-		// replace WhitelistedSubnetsKey flag
 		nodeConfig := node.GetConfig()
-		nodeConfig.ConfigFile, err = utils.SetJSONKey(nodeConfig.ConfigFile, config.WhitelistedSubnetsKey, whitelistedSubnets)
-		if err != nil {
-			return err
-		}
+
+		// delete node specific flag so as to use default one
+		delete(nodeConfig.Flags, config.WhitelistedSubnetsKey)
```

Comment:
The subsequent addNode is going to apply the network flag `ln.flags[config.WhitelistedSubnetsKey] = whitelistedSubnets`, unless the node flag is defined with an old value (eg from a previous subnet creation operation), in which case it will use the old value (node flag has precedence).
Two implementation where posible here:
1. update the whitelisted subnets value in ln.flags and remove the old value fron nodeConfig.flags
2. update nodeConfig.flags value (with or without changing the ln.flags value)
I opted for 1 trying to preserve an updated ln.flags. But can change to 2 is you prefer.

Reply:
OK I understand better now. This is not just some restart triggered from anywhere, but really only when we have to restart because there are new whitelisted subnets to add. I think we can leave this as-is then.